spec/codebreaker_spec.rb

```diff
@@ -0,0 +1,157 @@
+
+  describe '#create_user' do
+    context 'success' do
+      it 'creates user' do
+        expect(game.create_user('Username', difficulty)).to be_instance_of(Codebreaker::User)
+        expect(difficulty.attempts).to eq(attempts)
+        expect(difficulty.hints).to eq(hints)
+      end
+    end
+    context 'falsey' do
+      it 'creates user' do
+        expect { game.create_user(invalid_name, difficulty) }.to raise_error(Codebreaker::Errors::ClassValidError)
+        expect { game.create_user(' ', difficulty) }.to raise_error(Codebreaker::Errors::MinLengthError)
+        expect { game.create_user(invalid_name, difficulty) }.to raise_error(Codebreaker::Errors::ClassValidError)
+      end
+    end
+  end
+
+  describe '#create_difficulty' do
+    context 'success' do
+      it 'should have choose right name difficulty' do
+        expect do
+          game.create_difficulty(name).not_to raise_error(Codebreaker::Errors::InvalidDifficultyError)
+        end
+      end
+      it 'should have choose right attempts' do
+        expect(difficulty.attempts).to eq(15)
+      end
+
+      it 'should have choose right hints' do
+        expect(difficulty.hints).to eq(2)
+      end
+      context 'falsey' do
+        it 'enter invalid difficulty' do
+          expect do
+            game.create_difficulty(invalid_name).to raise_error(Codebreaker::Errors::InvalidDifficultyError)
+          end
+        end
+      end
+    end
+
+    describe '#take_attempt' do
+      context 'guess is valid' do
+        it '#take_attempt not change @attempts value if called for the first time' do
+          game.stub(:take_attempt)
+          expect { game.take_attempt(:win?) }.not_to(change { game.instance_variable_get(:@attempts) })
+        end
+      end
+      context 'true' do
+        keys = %w[hints attempts secret_code]
+
+        keys.each do |key|
+          it "return value should include :#{key}" do
+            game.stub(:take_attempt)
+            subject.instance_variable_set(:@secret_code, '1234')
+            expect(game.take_attempt([:"#{key}"])).to be_nil
+          end
+        end
+        context 'guess is invalid' do
+          it 'returns nil' do
+            game.stub(:take_attempt)
+            expect(game.take_attempt('invalid_code')).to be_falsey
+          end
+        end
+      end
+
+      describe '#take_hint' do
+        let(:user) { double('Codebreaker::User') }
+        let(:hints) { [1, 2, 3, 4] }
+        before do
+          allow(game).to receive(:user).and_return(user)
+          allow(user).to receive(:hints?).and_return(true)
+          allow(user).to receive(:hint).and_return(true)
+          game.instance_variable_set(:@hints, hints)
+        end
+
+        it '' do
+          [1, 2, 3, 4].reverse.each do |hint|
+            expect(game.take_hint).to eq hint
+          end
+        end
+      end
+    end
+
+    describe '#win?' do
+      before do
+        game.instance_variable_set(:@secret_code, [1, 2, 3, 4])
```

Comment:
just game.instance_variable_set(:@status, Game::STATUS_WIN) (or LOSE) 
and then check true or false in test